### ISSUE TITLE
Call for YOLOv3 support 

### ISSUE THREAD

**Issue by tanzhenyu** [keras-team/keras-cv#937]
**Short Description**
YOLOv3 is considered to be widely used, excellent tradeoff between performance and speed, even compared with YOLO5 / YOLO-X / YOLO-E.

**Papers**
https://arxiv.org/pdf/1804.02767.pdf

**Existing Implementations**
Publicly there are some really popular repos, such as [this](https://github.com/qqwweee/keras-yolo3), and [this](https://github.com/tensorflow/models/tree/master/official/projects/yolo). However, the one we provide here should work with GPUs and TPUs, be compatible with KerasCV's FasterRCNN and RetinaNet API.

**Other Information**
Being an anchor based model, we'd like to see this model to rely on some of our existing components, such as anchor generation, label matching, etc. A training script that trains the model to a *reasonable* state is required, however `model.compile` and `model.fit` is not required in the first PR, in the long run `model.fit` should be used in the training script.

**Comment by tanzhenyu** [comment]
@qqwweee

**Comment by tanzhenyu** [comment]
> > There is no existing implementation within TF ecosystem.
> 
> Are you sure?
> 
> https://github.com/tensorflow/models/tree/master/official/projects/yolo#models-in-the-library

I'm not sure if this one is maintained anymore (also missing spp version)

**Comment by bhack** [comment]
> I'm not sure if this one is maintained anymore (also missing spp version)

Probably it was not  pre-trained in that flavour but I see the layers and test:
https://github.com/tensorflow/models/blob/master/official/projects/yolo/modeling/layers/nn_blocks.py#L1128
https://github.com/tensorflow/models/blob/master/official/projects/yolo/modeling/layers/nn_blocks_test.py#L206

**Comment by bhack** [comment]
Also all the yolo folder passed from beta to official namespace just 1 month ago with this @arashwan's PR:
https://github.com/tensorflow/models/commit/9b7c561a997eed0c312a4dcaccd8e42168936fae

**Comment by tanzhenyu** [comment]
> > I'm not sure if this one is maintained anymore (also missing spp version)
> 
> Probably it was not pre-trained in that flavour but I see the layer and test: https://github.com/tensorflow/models/blob/master/official/projects/yolo/modeling/layers/nn_blocks.py#L1128 https://github.com/tensorflow/models/blob/master/official/projects/yolo/modeling/layers/nn_blocks_test.py#L206

The point is to ask for contribution for modeling and training script that can be consistent with our existing APIs

**Comment by bhack** [comment]
> The point is to ask for contribution for modeling and training script that can be consistent with our existing APIs

We have discussed this extensively at the launch  of this repo, so it  is not about if you want to implement this or not, it more about**bold** claims that doesn't correctly represent the status of the ecosystem.

So we have this model already in TF ecosystem "official" but we want to re-implement it in the Keras-cv style.
Then we could be inspired by refactoring some sub-components from model gardens, from third party repos or from scratch just reading the paper directly and talking with the original author or his official reference implementation. 

I really don't care how to go or if we adopt a mix of these approaches to be inspired as soon as we are going to maximizing the resources/hours spent on the contribution and to reproduce the results.

But at least we should keep a clear representation of OUR ecosystem even if there were different views among different ecosystem teams.

**Comment by tanzhenyu** [comment]
> > The point is to ask for contribution for modeling and training script that can be consistent with our existing APIs
> 
> We have discussed this extensively at the launch of this repo, so it is not about if you want to implement this or not, it is more about **bold** claims that don't correctly represent the status of the ecosystem.
> 
> So we have this model already in TF ecosystem "official" but we want to re-implement it in the Keras-cv style. Then we could be inspired by refactoring some sub-components from models garden, from third party repos or from scratch just reading the paper directly and talking with the original author or looking at his official reference implementation.
> 
> I really don't care how to go or if we adopt a mix of these approaches to be inspired as soon as we are going to maximizing the resources/hours spent on the contribution and to reproduce the results.
> 
> But at least we should keep a clear representation of OUR ecosystem even if there were different views among different ecosystem teams.

Correcting the request and cleaning up this so the contributors will not be confused.

**Comment by bhack** [comment]
Is totally removing comments our new community policy?
 
If we have resolved the topic re-editing the original description of the ticket we have other tools in GitHub to hide comments:

https://github.blog/2018-04-18-new-tools-for-open-source-maintainers/

So you just could hide comments as resolved,  instead in this way it seems that my discussion was banned.

**Comment by tanzhenyu** [comment]
> Is totally removing comments our new community policy?
> 
> If we have resolved the topic re-editing the original description of the ticket we have other tools in GitHub to hide comments:
> 
> https://github.blog/2018-04-18-new-tools-for-open-source-maintainers/
> 
> So you could just hide comments as resolved, instead in this way it seems that my discussion was banned.

This sounds like a much better tool. Done.

**Comment by innat** [comment]
@tanzhenyu It was requested https://github.com/keras-team/keras-cv/issues/622

**Comment by tanzhenyu** [comment]
> @tanzhenyu It was requested #622

Oh great, closing this to reflect that issue.
